apps/prairielearn/src/batched-migrations/20250318174636_course_instance_usages__submissions__backfill_v3.ts

```diff
@@ -0,0 +1,39 @@
+import { z } from 'zod';
+
+import { makeBatchedMigration } from '@prairielearn/migrations';
+import { loadSqlEquiv, queryAsync, queryOptionalRow } from '@prairielearn/postgres';
+
+const sql = loadSqlEquiv(import.meta.url);
+
+const START_DATE = '2025-02-15T00:00:00Z';
+const END_DATE = '2025-03-21T00:00:00Z';
+
+export default makeBatchedMigration({
+  async getParameters() {
+    // First delete old usage data
+    await queryAsync(sql.delete_old_usages, { START_DATE, END_DATE });
+
+    // Only backfill from submissions within the date range
+    const min =
+      (await queryOptionalRow(
+        sql.select_min_bound,
+        { START_DATE },
+        z.bigint({ coerce: true }).nullable(),
+      )) ?? 1n;
+    const max =
+      (await queryOptionalRow(
+        sql.select_max_bound,
+        { END_DATE },
+        z.bigint({ coerce: true }).nullable(),
+      )) ?? 1n;
```

Comment:
These values are allowed to be null, and specifically if the upper bound is null (no rows were found), the migration will immediately be marked as succeeded, which will keep this fast in dev mode.

Reply:
Good point, removed both `?? 1n` in b77b6095b6df30c10b22378bb69a0cdf90b21eb7